packages/kit/src/core/prerender/prerender.js

```diff
@@ -244,10 +244,13 @@ export async function prerender() {
 
 			const prerender = headers['x-sveltekit-prerender'];
 			if (prerender) {
-				const route_id = headers['x-sveltekit-routeid'];
-				const existing_value = prerender_map.get(route_id);
-				if (existing_value !== 'auto') {
-					prerender_map.set(route_id, prerender === 'true' ? true : 'auto');
+				const encoded_route_id = headers['x-sveltekit-routeid'];
+				if (encoded_route_id != null) {
```

Comment:
I might have just checked for truthiness would be a bit simpler to write as well as more robust to undefined or empty string. same below on line 312
```suggestion
				if (encoded_route_id) {
```

Reply:
That would be wrong because the root route is the empty string

Reply:
I sometimes wonder if route IDs should start with `/` for that reason. I've been caught out by that once or twice. Would be a breaking change, but a fairly minor one